colossalai/zero/shard_utils/flat_shard_strategy.py

```diff
@@ -0,0 +1,52 @@
+from tokenize import group
+import torch
+from colossalai.zero.shard_utils import BaseShardStrategy
+import torch.distributed as dist
+from typing import List, Optional
+from torch import Tensor
+from colossalai.zero.sharded_param.sharded_tensor import ShardedTensor
+from torch._utils import _flatten_dense_tensors, _unflatten_dense_tensors
+from colossalai.zero.sharded_model._zero3_utils import get_shard
+from colossalai.utils import get_current_device
+
+
+class FlatShardStrategy(BaseShardStrategy):
+
+    """
+    FlatShardStategy flattens all tensors and shards the 1D tensor.
+    """
+
+    def __init__(self, process_group: Optional[dist.ProcessGroup] = None) -> None:
+        super().__init__(process_group)
+
+    def shard(self, tensor_list: List[Tensor]) -> List[Tensor]:
+        flat_tensor = _flatten_dense_tensors(tensor_list)
+        new_tensor_list = _unflatten_dense_tensors(flat_tensor, tensor_list)
```

Comment:
why _unflatten_dense_tensors? will it turn flat_tensor back to tensor_list?

Reply:
Yes, flatten will create a new tensor. Need to unflatten and update the `.data` attribute of the original tensor.